src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -2168,4 +2168,126 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
     iterations: Int = 10000): KeyTable = {
     Skat(this, variantKeys, singleKey,  weightExpr, y, x, covariates, logistic, maxSize, accuracy, iterations)
   }
+
+  def trioMatrix(pedigree: Pedigree, completeTrios: Boolean): VariantSampleMatrix[Annotation, Annotation, Annotation] = {
+    if (!sSignature.isInstanceOf[TString])
+      fatal("trio_matrix requires column keys of type String")
+    requireUniqueSamples("trio_matrix")
+
+    val filteredPedigree = pedigree.filterTo(stringSampleIds.toSet)
+    val trios = if (completeTrios) filteredPedigree.completeTrios else filteredPedigree.trios
+    val nTrios = trios.length
+
+    val sampleIndices = sampleIds.zipWithIndex.toMap
+
+    val kidIndices = Array.fill[Int](nTrios)(-1)
+    val dadIndices = Array.fill[Int](nTrios)(-1)
+    val momIndices = Array.fill[Int](nTrios)(-1)
+    val kidIds = new Array[String](nTrios)
+
+    val memberAnnotationType = TStruct(
+      "id" -> TString(required = true),
+      "annotations" -> saSignature
+    )
+    val newColumnAnnotationType = TStruct(
+      "proband" -> memberAnnotationType,
+      "father" -> memberAnnotationType,
+      "mother" -> memberAnnotationType
+    )
+
+    val newSampleAnnotations = new Array[Annotation](nTrios)
+
+    var i = 0
+    while (i < nTrios) {
+      val t = trios(i)
+      val kidIndex = sampleIndices(t.kid)
+      kidIndices(i) = kidIndex
+      kidIds(i) = t.kid
+      val kidAnnotation = Row(t.kid, sampleAnnotations(kidIndex))
+
+      var dadAnnotation: Annotation = null
+      t.dad.foreach { dad =>
+        val index = sampleIndices(dad)
+        dadIndices(i) = index
+        dadAnnotation = Row(dad, sampleAnnotations(index))
+      }
+
+      var momAnnotation: Annotation = null
+      t.mom.foreach { mom =>
+        val index = sampleIndices(mom)
+        momIndices(i) = index
+        momAnnotation = Row(mom, sampleAnnotations(index))
+      }
+
+      newSampleAnnotations(i) = Row(kidAnnotation, dadAnnotation, momAnnotation)
+      i += 1
+    }
+
+    val newCellType = TStruct(
+      "proband" -> genotypeSignature,
+      "father" -> genotypeSignature,
+      "mother" -> genotypeSignature
+    )
+    val gsig = genotypeSignature
```

Comment:
If you're going to keep ``gsig``, change it to ``gSig`` and move it up a few lines so you can use it in newEntryType as well. Or (my pref) just use genotypeSignature throughout.